lib/elixir/lib/access.ex

```diff
@@ -137,10 +137,24 @@ defmodule Access do
 
   """
 
-  @type t :: list | map | nil | any
+  @type container :: keyword | struct | map | any
+  @type nil_container :: nil
+  @type t :: container | nil_container
   @type key :: any
   @type value :: any
 
+  @type get_fun(get_value, structure) ::
```

Comment:
The `structure` should be first argument to mimic all other APIs. Also please call it `data` as it is shorter.